libs/remote_storage/src/s3_bucket.rs

```diff
@@ -471,17 +468,7 @@ impl RemoteStorage for S3Bucket {
         // get the passed prefix or if it is not set use prefix_in_bucket value
         let list_prefix = prefix
             .map(|p| self.relative_path_to_s3_object(p))
-            .or_else(|| self.prefix_in_bucket.clone())
-            .map(|mut p| {
-                // required to end with a separator
-                // otherwise request will return only the entry of a prefix
-                if matches!(mode, ListingMode::WithDelimiter)
-                    && !p.ends_with(REMOTE_STORAGE_PREFIX_SEPARATOR)
-                {
-                    p.push(REMOTE_STORAGE_PREFIX_SEPARATOR);
-                }
-                p
-            });
+            .or_else(|| self.prefix_in_bucket.clone().map(|s| s + "/"));
```

Comment:
nit: should we use `REMOTE_STORAGE_PREFIX_SEPARATOR`?

Reply:
Fair point.  17460095d